### DISCUSSION TITLE
Why are Unlisted videos listed in Latest on homepage?

### DISCUSSION THREAD

**Discussion by u35253** [mediacms-io/mediacms#935] [Q&A]
Firstly, awesome software, so many features, so clean, so easy to set up and get going.  Truly great for the world.

**Question**: Can the vanilla MediaCMS installation be made to not list Unlisted videos in the Featured list, on the homepage?

**Background**: I had thought that the Unlisted videos would reasonably allow my user to upload a video, market it as unlisted, then share that link to the viewer, so that the viewer does not need to create an account and sign in to watch the video, all while avoiding creating obvious publicly-posted links to the video.

However, following this did allow any un-logged-in user to visit the homepage and view the "Unlisted" videos right there in the "Latest" list at the homepage of the MediaCMS instance!  I ended up setting these to Private, and that seems to have worked to hide them from the Latest list.

**Extension questions:**
I wonder a few things:

1. Is this a current expected behavior?
2. Is there an existing better practice to avoid this experience?

Hypothetically, this might occur if there is some cache/delay between uploading the video, marking as Unlisted, and then waiting for the Latest to update.

**Some ideas** are:

1. A fallback could be to **require all users to log in**, but that removes some convenience and adds quite a bit of friction.
2. Another option might be to **remove the "Latest" feed from the home page**, at least so the videos do not obviously leak there.  (Although, it could be nice to keep this around.)
3. Finally, maybe there is a way to "Mark as Unlisted" **before the video gets published publicly**.  Automatically publishing all new videos as Public seems like an awkward flow, and appears to be what I see.  Although, I can see that this is a sane default, if most memberships are "all or nothing".

**Aside:** Finally, if this topic was discussed elsewhere, apologies.  I did try searching the Issues and Discussions, and the web, a bit.  (I assume this might be fixable with customization/consulting, but thought I'd ask the community here, first.). There was a similar discussion about delaying publication, which seemed to indicate there should be a way to set an uploaded video to unlisted, but my uploads all seem to get set as public initially.

**Gratitude** Thanks again for making this.

**Comment by u35253** [comment]
Answering my own question, the answer appears to be yes, and the reason appears to have been because in `setttings.py`, the `PORTAL_WORKFLOW` variable had been set to "public", resulting in a public default visibility for newly uploaded media.

By changing the `PORTAL_WORKFLOW` value to "unlisted", I believe I see the anticipated behavior now.

I am okay with defaulting to one setting (unlisted), then manually setting public anything that should be public.

In my test, after making this change, the new upload no longer appears in the Homepage (e.g., on the Latest list) or in the History page.  This change seems to successfully circumvent that unwanted behavior, for this use case, AFAICT.

**Comment by u35253** [comment]
So, I take back the immediately preceding comment, somewhat.  In my experience, an un-logged-in user can still go to the History page, and see the Unlisted videos right there, then go to them, and play them.  This only seems to happen sporadically, though.  Sometimes, going to the History page shows 0 videos as expected.  But sometimes it shows the Unlisted videos.  And the playback pages that work do have the Unlisted tag on that page.  No user is logged in, but History page lists those Unlisted video still (in some fresh browser sessions).

I seem to have the best (only?) luck listing Unlisted videos on the History page by using the Brave browser, logged out, but not in Private Window mode.  For me right now, Firefox will not show Unlisted videos on the History page, and neither does Brave using a Private Window.

I would wonder if the listed videos are stored locally in the browser session somehow despite being logged out, but I had not viewed these videos in that browser before, or been logged in recently at all.

I don't understand the issue or the resolution at all at this point, so I hesitate to call it a bug just yet.  Maybe those "original" problem videos can be deleted, and maybe new uploads going forward will get better protected somehow by the `PORTAL_WORKFLOW` setting.  To be continued, I suppose.  Any input is welcome, of course.

**Comment by u35253** [comment]
Answering again my own followup question about "Unlisted videos listed in the History of un-logged-in browser sessions": the reason for being listed in the History is because that browser was used to visit those items already.

Visiting on a fresh browser session does not result in the Unlisted videos showing in History.

Returning to the History page (while logged out) does list the Unlisted items, but only if they were already visited (by that browser "session", which can persist across closing and reopening the browser application).

Commenting here in case it clears things up for anyone who may the same question in the future.

**Reply by nybhtw** [reply]
Funny how you experienced this.  I changed the settings.py as you suggested, and went to the site with a completely different browser (that has never been to the site), and it still shows under Home, Recommended, and Latest.

New browser, tried on a different PC, and they still show in the Latest feed.

My actual original goal was to self-host videos, unlisted, and have the videos embedded on different pages on my website.  I don't want anyone to be able to go to the hosted site to find these.

Love to find a different way to get this feature to work.

**Reply by u35253** [reply]
@nybhtw, did you stop and restart mediacms after changing settings.py?
